src/bloqade/lanes/layout/arch.py

```diff
@@ -36,6 +36,10 @@ class ArchSpec(Generic[SiteType]):
     """tuple of all words in the architecture. words[i] gives the word at word address i."""
     zones: tuple[tuple[int, ...], ...]
     """A tuple of zones where a zone is a tuple of word addresses and zone[i] gives the ith zone."""
+    measurement_zones: frozenset[int]
```

Comment:
What's unclear to me is translation between "zone" and "mode". Seems this conversion is pretty arbitrary because mode in flair only sees the SLM site coordinates, and we can get the information from `Word.all_positions()` -- they can be mapped via physical coordinates. 

Currently this mapping is implicit and on responsibility of a developer who writes rewrite rules. This also makes the rewrite rule not portable. I would define data structure to manage this mapping in this spec to make it more explicit.